tests/lib/rules/no-lonely-if.js

```diff
@@ -26,11 +27,105 @@ ruleTester.run("no-lonely-if", rule, {
     ],
 
     // Examples of code that should trigger the rule
-    invalid: [{
-        code: "if (a) {;} else { if (b) {;} }",
-        errors: [{
-            message: "Unexpected if as the only statement in an else block.",
-            type: "IfStatement"
-        }]
-    }]
+    invalid: [
+        {
+            code: "if (a) {;} else { if (b) {;} }",
+            output: "if (a) {;} else if (b) {;}",
+            errors
+        },
+        {
+            code:
+            "if (a) {\n" +
+            "  foo();\n" +
+            "} else {\n" +
+            "  if (b) {\n" +
+            "    bar();\n" +
+            "  }\n" +
+            "}",
+            output:
+            "if (a) {\n" +
+            "  foo();\n" +
+            "} else if (b) {\n" +
+            "    bar();\n" +
+            "  }",
+            errors
+        },
+        {
+            code:
+            "if (a) {\n" +
+            "  foo();\n" +
+            "} else /* comment */ {\n" +
+            "  if (b) {\n" +
+            "    bar();\n" +
+            "  }\n" +
+            "}",
+            output:
+            "if (a) {\n" +
+            "  foo();\n" +
+            "} else if (b) /* comment */ {\n" +
```

Comment:
At the moment, the fix is only aborted if there is a comment in one of these two places:

1. Between the `{` that opens `else`, and the `if` keyword
1. Between the `}` that closes `if`, and the `}` that closes `else`

These "comment-forbidden" locations were chosen because they actually get removed from the source code when the `else` and `if` are combined.

This comment is not in one of those locations, so the fixer decides that it can fix it safely (by putting `if (b)` before it, but otherwise leaving it intact).